content/automate/ManageIQ/System/CommonMethods/MiqAe.class/__methods__/weightedupdatestatus.rb

```diff
@@ -30,8 +30,8 @@ def reconcile_states_percent(path, states)
             end
 
             def create_cleanup_request(task)
-              sm_uri = @handle.object['cleanup_state_machine']
-              if sm_uri.present? && task.get_option(:cleanup_request_id).absent?
+              sm_uri = @handle.root['cleanup_state_machine']
```

Comment:
@fdupont-redhat Can you describe this change in more detail?  How is 'cleanup_state_machine' being resolved into the automate workspace?

cc @mkanoor

Reply:
`cleanup_state_machine` is an attribute of the main state machine `/Transformation/StateMachines/VMTransformation/Transformation` [1]. At first, I thought it would be an attribute of `$evm.object`, but dumping both `$evm.object` and `$evm.root` showed it was only available as an attribute of `$evm.root`. I also saw that it persists in the embedded state machines, which is good.

[1] https://github.com/fdupont-redhat/manageiq-content/blob/865897e21452a423c6769d869af3eb256ef4774d/content/automate/ManageIQ/Transformation/StateMachines/VMTransformation.class/__class__.yaml#L14-L33

Reply:
@fdupont-redhat 
The cleanup_state_machine is in the first state machine that starts which might be the root object. Then you go into nested state machines and each of the state machines ends up calling the same weightedupdatestatus method, if you are in the lower level state machines that cleanup_state_machine will not percolate down into the individual state machines, they would have their own $evm.object which would be the child state machine. You would have to check the $evm.root for this attribute or check if the weighted_status is being called from the correct state machine.

Reply:
Indeed cleanup_state_machine is available at the lowest level state machine, which creates the cleanup request. The parent state machines will also fail, but won't create additional requests because we check if a request already exists.